pennylane/ops/qubit/observables.py

```diff
@@ -235,6 +270,29 @@ def label(self, decimals=None, base_label=None):
         basis_string = "".join(str(int(i)) for i in self.parameters[0])
         return f"|{basis_string}⟩⟨{basis_string}|"
 
+    @staticmethod
+    def compute_matrix(basis_state):
+        """Canonical matrix representation of this operator.
+
+        Args:
+            basis_state (Iterable): basis state to project on
+
+        Returns:
+            array: canonical matrix
+
+        **Example**
+
+        >>> qml.Projector.compute_matrix([0, 1])
+        [[0. 0. 0. 0.]
+         [0. 1. 0. 0.]
+         [0. 0. 0. 0.]
+         [0. 0. 0. 0.]]
+        """
+        m = np.zeros((2 ** len(basis_state), 2 ** len(basis_state)))
+        idx = int("".join(str(i) for i in basis_state), 2)
+        m[idx, idx] = 1
```

Comment:
added this one from scratch, Projectors didn't have a matrix representation before